src/NServiceBus.AzureStoragePersistence.SagaDeduplicator/Extensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace NServiceBus.AzureStoragePersistence.SagaDeduplicator
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.WindowsAzure.Storage.Table;
+
+    public static class Extensions
+    {
+        public static EdmType GetPropertyType(this CloudTable table, string propertyName)
+        {
+            var query = new TableQuery
+            {
+                TakeCount = 1
+            };
+            EntityProperty propertyValue;
+            var entity = table.ExecuteQuery(query).SingleOrDefault();
```

Comment:
I've applied scanning across the whole table to find the property type in case of nulls.